File: python/compute_parrallel.py
import time 
from multiprocessing import Pool, cpu_count

def compute(n: int)->int:
    s=0
    for i in range(n):
        s+=i*i
    return s

if __name__=="__main__":
    N=10000000
    WORKERS=cpu_count()

    t1=time.time()
    with Pool(WORKERS) as p:
        results=p.map(compute, [N]*WORKERS)

    elapsed=time.time()-t1

    print("Result: ", sum(results))
    print("Elapsed (python parrallel):", elapsed)

File: rust/hello/src/bin/compute_parrallel.rs
use rayon::prelude::*;
use std::time::Instant;

fn compute(n: u128) -> u128 {
    (0..n).map(|i| i * i).sum()
}

fn main() {
    let n=10000000;
    let workers=num_cpus::get();

    let start=Instant::now();

    let result: u128 = (0..workers).into_par_iter().map(|_| compute(n)).sum();
    let elapsed=start.elapsed();

    println!("Result: {}", result);
    println!("Elapsed (rust parallel): {:?}", elapsed);

}